### ISSUE TITLE
Flutter 2.10 SurfaceAndroidWebView crash Android

### ISSUE THREAD

**Issue by EArminjon** [flutter/flutter#97732]
## Steps to Reproduce

1. flutter create .
2. replace code inside main.dart by mine
3. replace code inside pubspec.yaml by mine
4. set minSdkVersion to 21 (android/app/build.gradle)
5. flutter pub get
6. flutter pub run
7. go to settings page 
8. go the button called 'Notifications settings'
9. Honor the crash 

**Expected results:** 
No crash.

**Actual results:** 
A crash

<details>
<summary>Code sample</summary>

```dart
import 'package:app_settings/app_settings.dart';
import 'package:flutter/material.dart';
import 'package:webview_flutter/webview_flutter.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  final PageController controller = PageController();
  int index = 0;

  @override
  void dispose() {
    controller.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: PageView(
        onPageChanged: (int page) {
          setState(() => index = page);
        },
        controller: controller,
        physics: const NeverScrollableScrollPhysics(),
        children: const [
          WebViewPage(),
          SettingsPage(),
        ],
      ),
      bottomNavigationBar: BottomNavigationBar(
        currentIndex: index,
        onTap: (int page) {
          setState(() => controller.jumpToPage(page));
        },
        items: const [
          BottomNavigationBarItem(
            icon: Icon(Icons.web),
            label: "Webview",
          ),
          BottomNavigationBarItem(
            icon: Icon(Icons.settings),
            label: "Settings",
          ),
        ],
      ),
    );
  }
}

class WebViewPage extends StatefulWidget {
  const WebViewPage({Key? key}) : super(key: key);

  @override
  _WebViewPageState createState() => _WebViewPageState();
}

class _WebViewPageState extends State<WebViewPage> {
  @override
  void initState() {
    WebView.platform = SurfaceAndroidWebView();
    super.initState();
  }

  @override
  Widget build(BuildContext context) {
    return const WebView(
      initialUrl: "https://flutter.dev/",
    );
  }
}

class SettingsPage extends StatefulWidget {
  const SettingsPage({Key? key}) : super(key: key);

  @override
  _SettingsPageState createState() => _SettingsPageState();
}

class _SettingsPageState extends State<SettingsPage> {
  @override
  Widget build(BuildContext context) {
    return ListView(
      children: [
        ListTile(
          dense: true,
          contentPadding: EdgeInsets.zero,
          onTap: () => AppSettings.openNotificationSettings(),
          title: const Text("Notifications settings"),
          trailing: const Icon(Icons.chevron_right, color: Colors.grey),
        ),
      ],
    );
  }
}
```

```yaml
name: test
description: A new Flutter project.
publish_to: 'none' # Remove this line if you wish to publish to pub.dev
version: 1.0.0+1
environment:
  sdk: ">=2.16.0 <3.0.0"
dependencies:
  flutter:
    sdk: flutter
  cupertino_icons: ^1.0.2
  webview_flutter: 3.0.0
  app_settings: 4.1.1
dev_dependencies:
  flutter_test:
    sdk: flutter
  flutter_lints: ^1.0.0
flutter:
  uses-material-design: true
```

</details>

<details>
  <summary>Logs</summary>

```
Performing hot reload...
Syncing files to device Android SDK built for x86...
Reloaded 1 of 618 libraries in 458ms.
E/Surface ( 3561): getSlotFromBufferLocked: unknown buffer: 0xf5362790
E/chromium( 3561): [ERROR:aw_browser_terminator.cc(125)] Renderer process (3714) crash detected (code -1).
W/GooglePlayServicesUtil( 3561): Google Play Store is missing.
W/GooglePlayServicesUtil( 3561): Google Play Store is missing.
E/cr_PlatformSer-Internal( 3561): UsageReporting query failed
E/flutter ( 3561): [ERROR:flutter/fml/platform/android/jni_util.cc(211)] java.lang.RuntimeException: Methods marked with @UiThread must be executed on the main thread. Current thread: 1.raster
E/flutter ( 3561): 	at io.flutter.embedding.engine.FlutterJNI.ensureRunningOnMainThread(FlutterJNI.java:1405)
E/flutter ( 3561): 	at io.flutter.embedding.engine.FlutterJNI.destroyOverlaySurfaces(FlutterJNI.java:1174)
E/flutter ( 3561): 
F/flutter ( 3561): [FATAL:flutter/shell/platform/android/platform_view_android_jni_impl.cc(1559)] Check failed: fml::jni::CheckException(env). 
F/libc    ( 3561): Fatal signal 6 (SIGABRT), code -6 (SI_TKILL) in tid 3591 (1.raster), pid 3561 (om.example.test)
*** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
Build fingerprint: 'Android/sdk_phone_x86/generic_x86:11/RSR1.210210.001.A1/7193139:userdebug/dev-keys'
Revision: '0'
ABI: 'x86'
Timestamp: 2022-02-03 20:43:38+0000
pid: 3561, tid: 3591, name: 1.raster  >>> com.example.test <<<
uid: 10121
signal 6 (SIGABRT), code -6 (SI_TKILL), fault addr --------
Abort message: '[FATAL:flutter/shell/platform/android/platform_view_android_jni_impl.cc(1559)] Check failed: fml::jni::CheckException(env). 
'
    eax 00000000  ebx 00000de9  ecx 00000e07  edx 00000006
    edi f17cc81e  esi c4603ac0
    ebp f5db9b90  esp c4603a68  eip f5db9b99
backtrace:
      #00 pc 00000b99  [vdso] (__kernel_vsyscall+9)
      #01 pc 0005ad68  /apex/com.android.runtime/lib/bionic/libc.so (syscall+40) (BuildId: 6e3a0180fa6637b68c0d181c343e6806)
      #02 pc 00076511  /apex/com.android.runtime/lib/bionic/libc.so (abort+209) (BuildId: 6e3a0180fa6637b68c0d181c343e6806)
      #03 pc 01423bea  /data/app/~~jN8_Zuyx1u6wq27-kddn2Q==/com.example.test-9iMLWaJzI2BaJ89uott8kA==/lib/x86/libflutter.so (BuildId: bc2674e80bf793e9ed9bda34fe5506c5127bf8df)
      #04 pc 0144f149  /data/app/~~jN8_Zuyx1u6wq27-kddn2Q==/com.example.test-9iMLWaJzI2BaJ89uott8kA==/lib/x86/libflutter.so (BuildId: bc2674e80bf793e9ed9bda34fe5506c5127bf8df)
      #05 pc 01430f82  /data/app/~~jN8_Zuyx1u6wq27-kddn2Q==/com.example.test-9iMLWaJzI2BaJ89uott8kA==/lib/x86/libflutter.so (BuildId: bc2674e80bf793e9ed9bda34fe5506c5127bf8df)
      #06 pc 01c823c0  /data/app/~~jN8_Zuyx1u6wq27-kddn2Q==/com.example.test-9iMLWaJzI2BaJ89uott8kA==/lib/x86/libflutter.so (BuildId: bc2674e80bf793e9ed9bda34fe5506c5127bf8df)
      #07 pc 01c80a88  /data/app/~~jN8_Zuyx1u6wq27-kddn2Q==/com.example.test-9iMLWaJzI2BaJ89uott8kA==/lib/x86/libflutter.so (BuildId: bc2674e80bf793e9ed9bda34fe5506c5127bf8df)
      #08 pc 0185d247  /data/app/~~jN8_Zuyx1u6wq27-kddn2Q==/com.example.test-9iMLWaJzI2BaJ89uott8kA==/lib/x86/libflutter.so (BuildId: bc2674e80bf793e9ed9bda34fe5506c5127bf8df)
      #09 pc 0186caa1  /data/app/~~jN8_Zuyx1u6wq27-kddn2Q==/com.example.test-9iMLWaJzI2BaJ89uott8kA==/lib/x86/libflutter.so (BuildId: bc2674e80bf793e9ed9bda34fe5506c5127bf8df)
      #10 pc 0144c309  /data/app/~~jN8_Zuyx1u6wq27-kddn2Q==/com.example.test-9iMLWaJzI2BaJ89uott8kA==/lib/x86/libflutter.so (BuildId: bc2674e80bf793e9ed9bda34fe5506c5127bf8df)
      #11 pc 0145041b  /data/app/~~jN8_Zuyx1u6wq27-kddn2Q==/com.example.test-9iMLWaJzI2BaJ89uott8kA==/lib/x86/libflutter.so (BuildId: bc2674e80bf793e9ed9bda34fe5506c5127bf8df)
      #12 pc 0145032c  /data/app/~~jN8_Zuyx1u6wq27-kddn2Q==/com.example.test-9iMLWaJzI2BaJ89uott8kA==/lib/x86/libflutter.so (BuildId: bc2674e80bf793e9ed9bda34fe5506c5127bf8df)
      #13 pc 0145873c  /data/app/~~jN8_Zuyx1u6wq27-kddn2Q==/com.example.test-9iMLWaJzI2BaJ89uott8kA==/lib/x86/libflutter.so (BuildId: bc2674e80bf793e9ed9bda34fe5506c5127bf8df)
      #14 pc 0145876a  /data/app/~~jN8_Zuyx1u6wq27-kddn2Q==/com.example.test-9iMLWaJzI2BaJ89uott8kA==/lib/x86/libflutter.so (BuildId: bc2674e80bf793e9ed9bda34fe5506c5127bf8df)
      #15 pc 00019337  /system/lib/libutils.so (android::SimpleLooperCallback::handleEvent(int, int, void*)+39) (BuildId: ab4be013cda31e8c45d48aa23a89d0f8)
      #16 pc 0001a367  /system/lib/libutils.so (android::Looper::pollInner(int)+1127) (BuildId: ab4be013cda31e8c45d48aa23a89d0f8)
      #17 pc 00019e96  /system/lib/libutils.so (android::Looper::pollOnce(int, int*, int*, void**)+118) (BuildId: ab4be013cda31e8c45d48aa23a89d0f8)
      #18 pc 000133f5  /system/lib/libandroid.so (ALooper_pollOnce+101) (BuildId: f98006d4d736575082b552ff24b66f6c)
      #19 pc 0145867f  /data/app/~~jN8_Zuyx1u6wq27-kddn2Q==/com.example.test-9iMLWaJzI2BaJ89uott8kA==/lib/x86/libflutter.so (BuildId: bc2674e80bf793e9ed9bda34fe5506c5127bf8df)
      #20 pc 014502da  /data/app/~~jN8_Zuyx1u6wq27-kddn2Q==/com.example.test-9iMLWaJzI2BaJ89uott8kA==/lib/x86/libflutter.so (BuildId: bc2674e80bf793e9ed9bda34fe5506c5127bf8df)
      #21 pc 0144fe02  /data/app/~~jN8_Zuyx1u6wq27-kddn2Q==/com.example.test-9iMLWaJzI2BaJ89uott8kA==/lib/x86/libflutter.so (BuildId: bc2674e80bf793e9ed9bda34fe5506c5127bf8df)
      #22 pc 01455525  /data/app/~~jN8_Zuyx1u6wq27-kddn2Q==/com.example.test-9iMLWaJzI2BaJ89uott8kA==/lib/x86/libflutter.so (BuildId: bc2674e80bf793e9ed9bda34fe5506c5127bf8df)
      #23 pc 000e6974  /apex/com.android.runtime/lib/bionic/libc.so (__pthread_start(void*)+100) (BuildId: 6e3a0180fa6637b68c0d181c343e6806)
      #24 pc 00078567  /apex/com.android.runtime/lib/bionic/libc.so (__start_thread+71) (BuildId: 6e3a0180fa6637b68c0d181c343e6806)
Lost connection to device.
```

```
```

```
[√] Flutter (Channel stable, 2.10.0, on Microsoft Windows [version 10.0.19044.1503], locale fr-FR)
    • Flutter version 2.10.0 at C:\Users\**********\Documents\Softwares\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 5f105a6ca7 (2 days ago), 2022-02-01 14:15:42 -0800
    • Engine revision 776efd2034
    • Dart version 2.16.0
    • DevTools version 2.9.2

[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at C:\Users\**********\AppData\Local\Android\sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: C:\Users\**********\.jdks\openjdk-15\bin\java
    • Java version OpenJDK Runtime Environment (build 15+36-1562)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.9.0)
    • Visual Studio at D:\dev
    • Visual Studio Community 2019 version 16.9.31025.194
    • Windows 10 SDK version 10.0.19041.0

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/windows#android-setup for detailed instructions).

[☠] IntelliJ IDEA Ultimate Edition (the doctor check crashed)
    X Due to an error, the doctor check did not complete. If the error message below is not helpful, please let us know
      about this issue at https://github.com/flutter/flutter/issues.
    X FormatException: Unexpected extension byte (at offset 5)
    • #0      _Utf8Decoder.convertSingle (dart:convert-patch/convert_patch.dart:1789:7)
      #1      Utf8Decoder.convert (dart:convert/utf.dart:351:42)
      #2      InputStream.readString (package:archive/src/util/input_stream.dart:207:30)
      #3      new ZipDirectory.read (package:archive/src/zip/zip_directory.dart:40:30)
      #4      ZipDecoder.decodeBuffer (package:archive/src/zip_decoder.dart:19:30)
      #5      ZipDecoder.decodeBytes (package:archive/src/zip_decoder.dart:14:12)
      #6      IntelliJPlugins._findPluginXml (package:flutter_tools/src/intellij/intellij.dart:130:44)
      #7      IntelliJPlugins._readPackageVersion (package:flutter_tools/src/intellij/intellij.dart:141:40)
      #8      IntelliJPlugins.validatePackage (package:flutter_tools/src/intellij/intellij.dart:63:35)
      #9      IntelliJValidator.validate (package:flutter_tools/src/intellij/intellij_validator.dart:103:15)
      #10     asyncGuard.<anonymous closure> (package:flutter_tools/src/base/async_guard.dart:111:32)
      #11     asyncGuard.<anonymous closure> (package:flutter_tools/src/base/async_guard.dart:109:18)
      #12     _rootRun (dart:async/zone.dart:1426:13)
      #13     _CustomZone.run (dart:async/zone.dart:1328:19)
      #14     _runZoned (dart:async/zone.dart:1861:10)
      #15     runZonedGuarded (dart:async/zone.dart:1849:12)
      #16     runZoned (dart:async/zone.dart:1780:12)
      #17     asyncGuard (package:flutter_tools/src/base/async_guard.dart:109:3)
      #18     Doctor.startValidatorTasks (package:flutter_tools/src/doctor.dart:205:9)
      #19     Doctor.diagnose (package:flutter_tools/src/doctor.dart:309:47)
      #20     DoctorCommand.runCommand (package:flutter_tools/src/commands/doctor.dart:50:48)
      #21     FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:1320:12)      
      <asynchronous suspension>
      #22     FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1161:27)   
      <asynchronous suspension>
      #23     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
      <asynchronous suspension>
      #24     CommandRunner.runCommand (package:args/command_runner.dart:209:13)
      <asynchronous suspension>
      #25     FlutterCommandRunner.runCommand.<anonymous closure>
      (package:flutter_tools/src/runner/flutter_command_runner.dart:281:9)
      <asynchronous suspension>
      #26     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
      <asynchronous suspension>
      #26     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
      <asynchronous suspension>
      #27     FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:229:5)
      <asynchronous suspension>
      #28     run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:62:9)
      <asynchronous suspension>
      #29     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
      <asynchronous suspension>
      #30     main (package:flutter_tools/executable.dart:94:3)
      <asynchronous suspension>


[√] Connected device (5 available)
    • Mi 9T (mobile)                     • c8d2a7bf      • android-arm64  • Android 11 (API 30)
    • Android SDK built for x86 (mobile) • emulator-5554 • android-x86    • Android 11 (API 30) (emulator)
    • Windows (desktop)                  • windows       • windows-x64    • Microsoft Windows [version 10.0.19044.1503]    • Ch
rome (web)                       • chrome        • web-javascript • Google Chrome 97.0.4692.99
    • Edge (web)                         • edge          • web-javascript • Microsoft Edge 97.0.1072.55
    ! Device emulator-5556 is offline.

[√] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 2 categories.
```

</details>

**Comment by EArminjon** [comment]
Issue detected on emulator and real devices.

**Comment by blasten** [comment]
Duplicate of https://github.com/flutter/flutter/issues/96661

**Comment by blasten** [comment]
This issue is fixed upstream, and will be cherry-picked into 2.10

**Comment by EArminjon** [comment]
cherry picked into 2.10, I already have 2.10 what should i do ?

**Comment by wuweijian1997** [comment]
2.8.1 or 2.10.1

**Comment by EArminjon** [comment]
Fixed in 2.10.1 confirmed.
